diff --git a/CMakeLists.txt b/CMakeLists.txt
index b1fb2c1..69de239 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -113,7 +113,7 @@ install(
   FILES
     "${CMAKE_CURRENT_BINARY_DIR}/TTKBaseConfig.cmake"
   DESTINATION
-    "${CMAKE_INSTALL_LIBDIR}/cmake/ttkBase"
+    "${CMAKE_INSTALL_LIBDIR}/cmake"
   )
 
 
@@ -153,7 +153,7 @@ if(TTK_BUILD_VTK_WRAPPERS)
     NAMESPACE
       ttk::vtk::
     DESTINATION
-      ${CMAKE_INSTALL_LIBDIR}/cmake/ttkVTK
+      ${CMAKE_INSTALL_LIBDIR}/cmake
     )
 
   configure_file(core/vtk/TTKVTKConfig.cmake.in TTKVTKConfig.cmake @ONLY)
diff --git a/core/vtk/TTKVTKConfig.cmake.in b/core/vtk/TTKVTKConfig.cmake.in
index 06d9a91..f6a83c7 100644
--- a/core/vtk/TTKVTKConfig.cmake.in
+++ b/core/vtk/TTKVTKConfig.cmake.in
@@ -9,6 +9,6 @@ else()
 endif()
 
 # Include the actual config for TTK Base
-include("${CMAKE_CURRENT_LIST_DIR}/../ttkBase/TTKBaseConfig.cmake")
+include("${CMAKE_CURRENT_LIST_DIR}/TTKBaseConfig.cmake")
 # And the actual targets for TTK VTK
 include("${CMAKE_CURRENT_LIST_DIR}/TTKVTKTargets.cmake")
