--- topology-tool-kit/CMake/config.cmake	Tue Jun 27 11:00:26 2023
+++ topology-tool-kit/CMake/config.cmake	Tue Jun 27 14:27:35 2023
@@ -120,7 +120,7 @@
   set_property(GLOBAL PROPERTY RULE_LAUNCH_COMPILE "${CMAKE_COMMAND} -E time")
 endif()
 
-set(TTK_SCRIPTS_PATH scripts/ttk
+set(TTK_SCRIPTS_PATH "${_VCPKG_INSTALLED_DIR}/x64-windows/share/scripts/ttk"
   CACHE PATH "Install path for TTK scripts"
   )
 mark_as_advanced(TTK_SCRIPTS_PATH)
