diff --git forkSrcPrefix/core/base/ripsPersistenceDiagram/ripserpy.cpp forkDstPrefix/core/base/ripsPersistenceDiagram/ripserpy.cpp
index 60a7bfcac27e2600a1ca9ddd782bcdb04d884191..7b455b2815303c6b88bef9743f51da7d639da81e 100644
--- forkSrcPrefix/core/base/ripsPersistenceDiagram/ripserpy.cpp
+++ forkDstPrefix/core/base/ripsPersistenceDiagram/ripserpy.cpp
@@ -93,13 +93,7 @@ static const size_t num_coefficient_bits = 8;
 
 // 1L on windows is ALWAYS 32 bits, when on unix systems is pointer size
 static const index_t max_simplex_index
-  = (__int128(1) << (8 * sizeof(index_t) - 1 - num_coefficient_bits)) - 1;
-
-namespace std {
-  inline std::string to_string(__int128) {
-    return "";
-  }
-} // namespace std
+  = (uintptr_t(1) << (8 * sizeof(index_t) - 1 - num_coefficient_bits)) - 1;
 
 void check_overflow(index_t i) {
   if
@@ -1079,4 +1073,4 @@ void ripser::ripser(std::vector<std::vector<value_t>> points,
       std::move(dist), dim_max, threshold, ratio, modulus);
     ripser.compute_barcodes(ph);
   }
-}
+}
\ No newline at end of file
