diff --git a/core/vtk/ttkIntegralLines/ttkIntegralLines.cpp b/core/vtk/ttkIntegralLines/ttkIntegralLines.cpp
index 8f4b0e490..a876994e5 100644
--- topology-tool-kit/core/vtk/ttkIntegralLines/ttkIntegralLines.cpp
+++ topology-tool-kit/core/vtk/ttkIntegralLines/ttkIntegralLines.cpp
@@ -425,14 +425,19 @@ int ttkIntegralLines::RequestData(vtkInformation *ttkNotUsed(request),
 #endif
 #endif
   // make the vtk trajectories
+#ifdef TTK_ENABLE_MPI
+  ttkTemplateMacro(
+    triangulation->getType(),
+    (getTrajectories<TTK_TT>(
+      domain, static_cast<TTK_TT *>(triangulation->getData()), integralLines,
+      globalVertexId, globalCellId, output)));
+#else
   ttkTemplateMacro(
     triangulation->getType(),
     (getTrajectories<TTK_TT>(
       domain, static_cast<TTK_TT *>(triangulation->getData()), integralLines,
-#ifdef TTK_ENABLE_MPI
-      globalVertexId, globalCellId,
-#endif
       output)));
+#endif
 
   return (int)(status == 0);
 }
